### PR TITLE
Prepare docs for Doctrine website

### PR DESCRIPTION
**Pull request by driehle**
Similar to doctrine/DoctrineModule#733, this PR prepares the docs for being rendered on the Doctrine website. The README has been split into multiple sections and converted from .md to .rst. A .doctrine-project.json is added.

Similar to and in line with:

- https://github.com/doctrine/DoctrineModule/pull/733
- https://github.com/doctrine/DoctrineORMModule/pull/672
- https://github.com/doctrine/DoctrineMongoODMModule/pull/230

### CODE REVIEW

File: docs/en/basic-usage.rst
Comment (SenseException) [inline docs/en/basic-usage.rst]:
```suggestion
.. note::

   internally, for a given collection, if an array contains
```

File: docs/en/performance-considerations.rst
Comment (SenseException) [inline docs/en/performance-considerations.rst]: Not sure why those are used here while not in the `If you are not comfortable with Fieldsets` paragraph. Did you plan to use a note here?

**Review by SenseException — CHANGES_REQUESTED** [review]

File: docs/en/basic-usage.rst
Comment (driehle) [reply docs/en/basic-usage.rst]: Fixed

File: docs/en/performance-considerations.rst
Comment (driehle) [reply docs/en/performance-considerations.rst]: Probably a left-over from the pandoc conversion. I added a note there.

**Review by SenseException — APPROVED** [review]

**Review by TomHAnderson — APPROVED** [review]
I'm going to approve this **for now**.  But there are a number of grammatical errors throughout the docs which should be corrected.  I'm willing to do a PR on the docs as a follow-up to this PR to address them.  I'm more concerned about getting this in place on the website.

Please create an issue, assigned to me, to review the docs' English as a follow-up to approving this PR.

**Comment by TomHAnderson** [comment]
If you want me to push a PR to this PR before you merge, I'm OK with that.  I'll leave the decision for when to merge up to @driehle

**Comment by driehle** [comment]
@TomHAnderson I'll merge this for now as this will likely enable me to render `doctrine-website` locally to see if the basic integration is all set up.

**Pull request merged**
